### PR TITLE
Documented preprocessing options and added an end-to-end example

### PR DESCRIPTION
**Pull request by rgesteve**
**Description**: Documents and provides an end-to-end example of how to carry out calibration using preprocessing, both via a built-in preprocess function, or through an offline process that requires more manual intervention.

**Motivation and Context**
- Calibration for quantization is a fairly elaborate process with a non-trivial amount of knobs, that may be daunting for a developer unfamiliar with the concepts, this documentation aims to provide and easy-to-follow guide to remove some of the difficulties.

### CODE REVIEW

File: onnxruntime/python/tools/quantization/README.md [hunk 1]
```diff
@@ -1,12 +1,52 @@
...
+
+To add the current set of available preprocessing functions, you need to add a Python function to the `data_preprocess.py` file.  Such a function needs to take a filename and the (width, height) of an image as integers, and return a `numpy` `ndarray`.
+
+To expose it a new preprocessing function to the command line, you'll need to "register" it by adding it to the name/function mapping maintained as a dictionary in the `set_preprocess` function in `data_preprocess.py`.
+
+### End-to-end example
+
+Say you want to quantize a model `/models/mymodel.onnx` that has been trained with a subset of the ILSVRC ImageNet dataset.  In what follows it is assumed that you have either the dataset or a sample you deem significant to use as a calibration set --  one way to get this is to use [a script from PaddlePaddle](https://github.com/PaddlePaddle/models/blob/develop/fluid/image_classification/data/ILSVRC2012/download_imagenet2012.sh) to fetch it.  Assume that the images from the dataset are in the `/tmp/data/ilsvrc12` directory (note you don't need the labels to quantize, but they are useful to validate/verify the results of the quantization).  You can then run
```
Comment (jywu-msft) [inline onnxruntime/python/tools/quantization/README.md]: that paddlepaddle link doesn't work.
i think it got moved to https://github.com/PaddlePaddle/models/blob/develop/PaddleCV/image_classification/data/ILSVRC2012/download_imagenet2012.sh

Comment (rgesteve) [reply onnxruntime/python/tools/quantization/README.md]: Thanks!  let me revise, and, hopefully, find out a more stable way to get an imagenet (sub) set

Comment (askhade) [reply onnxruntime/python/tools/quantization/README.md]: You can also use the instructions from mlperf repo... or simply point to the instructions in mlperf repo

Comment (rgesteve) [reply onnxruntime/python/tools/quantization/README.md]: Sorry for the delay, Ashiwini/George... now referencing https://github.com/mlperf/inference/blob/master/v0.5/classification_and_detection/GettingStarted.ipynb ... would you guys please take a look?

**Pull request merged**